CitationStyleLanguagePlugin.inc.php

```diff
@@ -358,14 +464,38 @@ public function getCitation($request, $article, $citationStyle = 'apa', $issue =
 					$currentAuthor->family = htmlspecialchars($author->getLocalizedFamilyName());
 					$currentAuthor->given = htmlspecialchars($author->getLocalizedGivenName());
 				}
-				$citationData->author[] = $currentAuthor;
+				$userGroup = $author->getUserGroup();
+				if (null !== $userGroup) {
+					switch ($userGroup->getId()) {
+						case $this->getEditorGroup($context->getId()):
+							if (!isset($citationData->editor)) {
+								$citationData->editor = array();
+							}
+							$citationData->editor[] = $currentAuthor;
+							break;
+						case $this->getTranslatorGroup($context->getId()):
+							if (!isset($citationData->translator)) {
+								$citationData->translator = array();
+							}
+							$citationData->translator[] = $currentAuthor;
+							break;
+						case $this->getAuthorGroup($context->getId()):
```

Comment:
Each time you call one of the methods to get the editor/translator/author group, you'll make a separate request to the database. I think with a `switch` statement that means it could run through the checks 3 times per author.

Can you get the editor/translator/author groups before the `switch` statement, so that each `case` line doesn't make a new run to the database?